src/Samples/DSSample.cpp

```diff
@@ -0,0 +1,104 @@
+#include "../../headers/Samples/DSSample.h"
+
+void DSSample::add(const int &value) {
+    this->baseAdd(value);
+}
+
+void DSSample::add(const float &value) {
+    this->baseAdd(value);
+}
+
+void DSSample::add(const double &value) {
+    this->baseAdd(value);
+}
+
+void DSSample::add(const bool &value) {
+    this->baseAdd(value);
+}
+
+void DSSample::add(const long &value) {
+    this->baseAdd(value);
+}
+
+void DSSample::addCategorical(const int &value, const std::vector<int> &possibleValues) {
+    this->baseAddCategorical(value, possibleValues);
+}
+
+void DSSample::addCategorical(const float &value, const std::vector<float> &possibleValues) {
+    this->baseAddCategorical(value, possibleValues);
+}
+
+void DSSample::addCategorical(const double &value, const std::vector<double> &possibleValues) {
+    this->baseAddCategorical(value, possibleValues);
+}
+
+void DSSample::addCategorical(const bool &value, const std::vector<bool> &possibleValues) {
+    this->baseAddCategorical(value, possibleValues);
+}
+
+void DSSample::addCategorical(const long &value, const std::vector<long> &possibleValues) {
+    this->baseAddCategorical(value, possibleValues);
+}
+
+void DSSample::addCategorical(const char &value, const std::vector<char> &possibleValues) {
+    this->baseAddCategorical(value, possibleValues);
+}
+
+void DSSample::reinitData() {
+    this->data.clear();
+}
+
+double* DSSample::getData() {
+    return &this->data[0];
```

Comment:
getData should copy the contents of the vector in a new array or an array passed in parameter for performance purposes